google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2423,6 +2471,48 @@ private void releaseSession(PooledSession session, Position position) {
     }
   }
 
+  private boolean isUnbalanced(PooledSession session) {
```

Comment:
Can we add `@VisibleForTesting` and add a few more unit tests for this method? It's doing a bunch of things which is currently untested with the existing tests.

Reply:
Yeah, good point. I've changed it a bit so we have a static method that does the actual calculation. That method is easier to test than one that depends on all the state fields in the SessionPool class. And I've added different tests for that method.